bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java

```diff
@@ -287,20 +287,17 @@ public String getLabel(Widget w) {
         if (itemName != null) {
             State state = null;
             String formatPattern = null;
-            int indexOpenBracket = label.indexOf("[");
-            int indexCloseBracket = label.indexOf("]");
 
             try {
                 Item item = getItem(itemName);
-                if (!label.contains("[") && item.getStateDescription() != null
+                if (getFormatPattern(label)==null && item.getStateDescription() != null
                         && item.getStateDescription().getPattern() != null) {
                     label = label + " [" + item.getStateDescription().getPattern() + "]";
                 }
 
-                if (label.contains("[")) {
-                    indexOpenBracket = label.indexOf("[");
-                    indexCloseBracket = label.indexOf("]");
-                    formatPattern = label.substring(indexOpenBracket + 1, indexCloseBracket);
+                String updatedPattern = getFormatPattern(label);
```

Comment:
We could move this a line before the if statement and use the variable in the first test.
So no need to call the function twice. Sure, no problem at all, but I try to reduce the function calls if possible.